### PR TITLE
(CDAP-6847) Documented cross namespace dataset access

### PR DESCRIPTION
**Pull request by bdmogal**
Build: http://builds.cask.co/browse/CDAP-DQB114

Relevant pages:
[Cross Namespace Dataset Access](http://builds.cask.co/artifact/CDAP-DQB114/shared/build-1/Docs-HTML/3.5.1-SNAPSHOT/en/developers-manual/building-blocks/datasets/index.html#cross-namespace-dataset-access)
[Authorization Pushdown](http://builds.cask.co/artifact/CDAP-DQB114/shared/build-1/Docs-HTML/3.5.1-SNAPSHOT/en/admin-manual/security/authorization.html#authorization-policy-pushdown)

### CODE REVIEW

**Comment by awholegunch** [comment]
Revised and restructured: Quick Build: http://builds.cask.co/browse/CDAP-DQB114-3

Cross Namespace Dataset Access: http://builds.cask.co/artifact/CDAP-DQB114/shared/build-3/Docs-HTML/3.5.1-SNAPSHOT/en/developers-manual/building-blocks/datasets/overview.html#cross-namespace-dataset-access

File: cdap-docs/admin-manual/source/security/authorization.rst [hunk 1]
```diff
@@ -349,25 +349,30 @@ select users:
...
+
+This limitation has a larger implication when :ref:`Cross-namespace Dataset Access
+<cross-namespace-dataset-access>` is used. When accessing a dataset from a different
+namespace, CDAP currently presumes that the user accessing the dataset has been granted
+permissions on the dataset in the storage provider prior to accessing the dataset from
+CDAP. 
+
+For example, if a program in the namespace *ns1* tries to access a :term:`fileset` in the
+namespace *ns2*, the user running the program should be granted the appropriate (*READ*,
+*WRITE*, or both) privileges on the fileset. Additionally, the user need to be granted
```
Comment (bdmogal) [inline cdap-docs/admin-manual/source/security/authorization.rst]: need_s_

Comment (awholegunch) [reply cdap-docs/admin-manual/source/security/authorization.rst]: Fixed.

**Comment by awholegunch** [comment]
@Kicked off another Quick Build: http://builds.cask.co/browse/CDAP-DQB114-4

This fails, but unrelated to this PR. Otherwise, LGTM.

@anew Could you look at the [Revised Building Blocks: Datasets](http://builds.cask.co/artifact/CDAP-DQB114/shared/build-4/Docs-HTML/3.5.1-SNAPSHOT/en/developers-manual/building-blocks/datasets/index.html)? We have split the single file on _FileSets_ into three, as that makes it easier to digest: the left sidebar table of contents then reflects each section, rather than opening all of the three sections at the same time. (You can compare it with the [current](http://docs.cask.co/cdap/current/en/developers-manual/building-blocks/datasets/index.html).)

This also has the advantage that the first page (now the [Overview](http://builds.cask.co/artifact/CDAP-DQB114/shared/build-4/Docs-HTML/3.5.1-SNAPSHOT/en/developers-manual/building-blocks/datasets/overview.html)) has a table of contents that appears in the left sidebar, which helps in following and finding the structure.

**Comment by bdmogal** [comment]
LGTM

**Comment by bdmogal** [comment]
Squashed commits. Merging.

**Pull request merged**